bugfix: Tracker was being disabled after sending an analytics event

corycaywood opened the pull request:
# Description 
In 'Analytics' class, the  Measurement's URL field is being cast to a String, however this throws an error if the field is a URL instance. This would then trigger the fail-safe for the Tracker, and disable performance tracking. This commit uses 'String#valueOf' instead of casting

## Links
[SDKCF-758]

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

corycaywood committed:
bugfix: Tracker was being disabled after sending an analytics event

In 'Analytics' class, the  Measurement's URL field is being cast to a 'String', however this throws an error if the field is a 'URL' instance. This would then trigger the fail-safe for the Tracker, and disable performance tracking. This commit uses 'String#valueOf' instead of casting

NemoOudeis commented:
@corycaywood 
good catch! 👍
I think we should make the `Analytics#sendUrlMeasurement` method fail safe, wrapping it with try-catch. A failure in analytics tracking should never stop the tracking behavior in my opinion. Or do you see a valid case where we would want that?

corycaywood commented:
Yeah, that's a good idea. I'll update the PR.

corycaywood committed:
fix: Catch all Exceptions by Analytics#sendUrlMeasurement

This will prevent the Tracker from being disabled due to any errors from sending Analytics
